qubes/storage/__init__.py

```diff
@@ -134,8 +134,12 @@ def __init__(self, name, pool, vid,
         self.source = source
         #: Volume unique (inside given pool) identifier
         self.vid = vid
+        #: Asynchronous lock for cryptsetup operations
+        self._crypt_lock = asyncio.Lock()
         #: Asynchronous lock for @Volume.locked decorator
         self._lock = asyncio.Lock()
+        #: Name of the encrypted version of the volume
+        self.encrypted_name = None
```

Comment:
You set this only in `start_encrypted`, and then condition cleanup based on this variable. Restarting qubesd in the meantime will make it skip the cleanup :(

Reply:
Good catch!  The purpose of this variable is so that the cleanup code knows what name to use, but that is only an issue when restarting a volume, and only the LVM pool needs it.  The LVM pool can calculate that info on its own, though.